### PR TITLE
Add more diagrams and explanations to the JSON codecs guide

### PR DESCRIPTION
**Pull request by julienrf**
There is a lot that could be improved, like providing complete
examples, and detailing how to use `JsonEntities` alone, but
I believe this could be done in a separate PR.

### CODE REVIEW

File: documentation/manual/src/doc/guides/json-codecs.md [hunk 1]
```diff
@@ -2,25 +2,82 @@
...
+Doc(Do I want to publish documentation?)-->|yes| JsonSchemaEntities
+Doc -->|no| Codec(Do I want to implement both a client and a server?)
+Codec -->|yes| JsonEntitiesFromCodec
+Codec -->|no| JsonEntities
 
-~~~ scala src=../../../../../algebras/algebra/src/main/scala/endpoints/algebra/JsonEntities.scala#request-response-types
+style JsonSchemaEntities fill:#fff
+style JsonEntitiesFromCodec fill:#fff
+style JsonEntities fill:#fff
 ~~~
```
Comment (julienrf) [inline documentation/manual/src/doc/guides/json-codecs.md]: This diagram looks like the following:

questions

File: documentation/manual/src/doc/guides/json-codecs.md [hunk 2]
```diff
@@ -2,25 +2,82 @@
...
+and the remaining sections provide more details on how to use `JsonSchemaEntities`
+and `JsonEntitiesFromCodec`.
+
+## The `JsonEntities` hierarchy
+
+The following diagram shows the relations between the three aforementioned algebras,
+`JsonEntities`, `JsonEntitiesFromCodec`, and `JsonSchemaEntities`, and their relations
+with the other algebras:
+
+![](json-entities.svg)
```
Comment (julienrf) [inline documentation/manual/src/doc/guides/json-codecs.md]: That one looks like this:

hierarchy

**Comment by julienrf** [comment]
@fduraffourg, can I ask a review?

File: documentation/manual/src/doc/guides/json-codecs.md (hunk 1 above)
Comment (fduraffourg) [reply documentation/manual/src/doc/guides/json-codecs.md]: Ho... Nice!

**Comment by fduraffourg** [comment]
Cool! That makes a lot more sense to me now :)

**Pull request merged**